python/sanlock.c

```diff
@@ -953,7 +953,7 @@ py_get_lockspaces(PyObject *self __unused, PyObject *args, PyObject *keywds)
             goto exit_fail;
 
         /* fill the dictionary information: path */
-        if ((ls_value = PyString_FromString(lss[i].host_id_disk.path)) == NULL)
+        if ((ls_value = PyBytes_FromString(lss[i].host_id_disk.path)) == NULL)
```

Comment:
This is a path - does not belong to this patch - probably should move to the patch handling paths.

Reply:
we'd like py3 and py2 to compile and test at this point so we better let it be.
we'll have FSConverter patch later for paths and we'll deal with resource path as well.

Reply:
We have a patch for handling strings and bytes, this patch is about int/long handling.

Reply:
AFAIK its in this commit - creating PyBytes object out of cstring.
The other commit in this PR regarding cstring is for getting the inner chars represnetation of a Pystring/Pyunicode object, the other way around.

Reply:
This handles converting sanlock path (cstring) to python type (unicode).

Using bytes is wrong for path. If you don't wan to mix with the change handling path input, split a path handling path output.

Reply:
We can use Py_BuildValue("s", path) to convert the paths until we have a better way.

On py2:
s (string) [char *]
Convert a null-terminated C string to a Python object. If the C string pointer is NULL, None is used.

On py3:
s (str or None) [const char *]
Convert a null-terminated C string to a Python str object using 'utf-8' encoding. If the C string pointer is NULL, None is used.

Which is symmetric to what we accept as input for path.

This can also be used for converting numeric values:

   Py_BuildValue("K", offset);

py2:
K (int) [unsigned long long]
Convert a C unsigned long long to a Python integer object.

py3:
K (long) [unsigned PY_LONG_LONG]
Convert a C unsigned long long to a Python long integer object. Only available on platforms that support unsigned long long.

Reply:
Note also the magic {items} format:

{items} (dictionary) [matching-items]
Convert a sequence of C values to a Python dictionary. Each pair of consecutive C values adds one item to the dictionary, serving as key and value, respectively.

Can simplify a lot the code creating dicts and list or dicts. Here is an example from _ssl moduel:

```
return Py_BuildValue("{sisisi}", "x509", x509, "crl", crl, "x509_ca", ca);
```